air-quality-backend/src/air_quality/etl/in_situ/openaq_adapter.py

```diff
@@ -42,36 +47,74 @@ def _create_document(
     }
 
 
+def _create_measurement_value(measurement):
+    return {
+        "value": measurement["value"],
+        "unit": measurement["unit"],
+        "original_value": measurement["value"],
+        "original_unit": measurement["unit"],
+    }
+
+
 def combine_measurement(state, measurement):
     key = f"{measurement['location']}_{measurement['date']['utc']}"
     results = state["results"]
     if key not in results:
         results[key] = _create_document(
             measurement, state["city"], state["location_type"]
         )
-    measurement_value = measurement["value"]
+    measurement_value = _create_measurement_value(measurement)
     measurement_parameter = measurement["parameter"]
     measurement_parameter_key = required_pollutant_data[measurement_parameter]
     results[key][measurement_parameter_key.value] = measurement_value
     return state
 
 
-def transform(in_situ_data):
+def transform_city(city_data):
     formatted_dataset = []
-    for city_name, city_data in in_situ_data.items():
-        city = city_data["city"]
-        measurements_for_city = city_data["measurements"]
-        if len(measurements_for_city) > 0:
-            filtered_measurements = filter(
-                measurement_value_is_positive, measurements_for_city
-            )
-            grouped_measurements = reduce(
-                combine_measurement,
-                filtered_measurements,
-                {"city": city["name"], "location_type": city["type"], "results": {}},
-            )["results"]
-            formatted_dataset.extend(list(grouped_measurements.values()))
-        else:
-            logging.info(f"No in situ measurements found for {city_name}")
+    city = city_data["city"]
+    measurements_for_city = city_data["measurements"]
+    if len(measurements_for_city) > 0:
+        filtered_measurements = filter(
+            measurement_value_is_positive, measurements_for_city
+        )
+        grouped_measurements = reduce(
+            combine_measurement,
+            filtered_measurements,
+            {"city": city["name"], "location_type": city["type"], "results": {}},
+        )["results"]
+        formatted_dataset.extend(list(grouped_measurements.values()))
+    else:
+        logging.info(f"No in situ measurements found for {city['name']}")
 
     return formatted_dataset
+
+
+def enrich_with_forecast_data(city_data, forecast_data: ForecastData):
+    for in_situ_reading in city_data:
```

Comment:
Do you reckon it's worth creating a TypedDict for the in_situ_readings so we have some protection on what properties there are?

Reply:
I've added this in, let me know what you think